app/aggregates/deciding/decision.rb

```diff
@@ -85,8 +94,10 @@ def send_to_provider(user_id:, application_id:)
 
     # When the decision is created from a MAAT record
     on DraftCreatedFromMaat do |event|
-      update_from_maat(event.data.fetch(:maat_decision))
-
+      maat_attributes = event.data.fetch(:maat_decision)
+      @maat_id = maat_attributes.fetch(:maat_id)
+      @reference = maat_attributes.fetch(:reference)
+      update_from_maat(maat_attributes)
```

Comment:
Refactor so that `reference` and `maat_id` are only set when the draft is first created. (Storing the reference of the decision as stored in MAAT is now valuable for understanding the history of a decision.)